thirdparty/kpvcrlib/CMakeLists.txt

```diff
@@ -202,7 +202,7 @@ add_library(crengine SHARED ${CRENGINE_SOURCES})
 target_compile_features(crengine PRIVATE cxx_constexpr)
 
 target_link_libraries(crengine PRIVATE chmlib antiword)
-target_link_libraries(crengine PUBLIC ${THIRDPARTY_LIBS})
+target_link_libraries(crengine PRIVATE ${THIRDPARTY_LIBS})
```

Comment:
When playing with it, I was a bit unsure it was allowed to have 2 target_link_libraries.
So, just be sure you can open a png and a chm after the build :)

> Unfamiliar with CRe's API 

cre.cpp only uses bits declared in crengine include/* - no direct call to things in the libraries used by crengine.

Reply:
It'd have failed to link ;).

Reply:
And everything is indeed there:

```
┌─(niluje@illyria:pts/14)───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────(/var/tmp/niluje/koreader)─┐
└─(0.12:78%:16:41:%)── readelf -d koreader-android-arm-linux-androideabi-debug/koreader/libs/libcrengine.so                                                                                                                                                                                                                                            ──(Sat, Sep 03)─┘

Dynamic section at offset 0x21d600 contains 42 entries:
  Tag        Type                         Name/Value
 0x00000003 (PLTGOT)                     0x21ec14
 0x00000002 (PLTRELSZ)                   1984 (bytes)
 0x00000017 (JMPREL)                     0x591ec
 0x00000014 (PLTREL)                     REL
 0x00000011 (REL)                        0x44514
 0x00000012 (RELSZ)                      85208 (bytes)
 0x00000013 (RELENT)                     8 (bytes)
 0x6ffffffa (RELCOUNT)                   10253
 0x00000006 (SYMTAB)                     0x148
 0x0000000b (SYMENT)                     16 (bytes)
 0x00000005 (STRTAB)                     0x12528
 0x0000000a (STRSZ)                      160197 (bytes)
 0x00000004 (HASH)                       0x396f0
 0x00000001 (NEEDED)                     Shared library: [libz.so.1]
 0x00000001 (NEEDED)                     Shared library: [libzstd.so.1]
 0x00000001 (NEEDED)                     Shared library: [libfreetype.so.6]
 0x00000001 (NEEDED)                     Shared library: [libharfbuzz.so.0]
 0x00000001 (NEEDED)                     Shared library: [libfribidi.so.0]
 0x00000001 (NEEDED)                     Shared library: [libunibreak.so.3]
 0x00000001 (NEEDED)                     Shared library: [libutf8proc.so.2]
 0x00000001 (NEEDED)                     Shared library: [libjpeg.so.8]
 0x00000001 (NEEDED)                     Shared library: [libpng16.so.16]
 0x00000001 (NEEDED)                     Shared library: [libwebp.so.7]
 0x00000001 (NEEDED)                     Shared library: [libwebpdemux.so.2]
 0x00000001 (NEEDED)                     Shared library: [libgnustl_shared.so]
 0x00000001 (NEEDED)                     Shared library: [libm.so]
 0x00000001 (NEEDED)                     Shared library: [libc.so]
 0x0000000e (SONAME)                     Library soname: [libcrengine.so]
 0x0000001a (FINI_ARRAY)                 0x21a774
 0x0000001c (FINI_ARRAYSZ)               8 (bytes)
 0x00000019 (INIT_ARRAY)                 0x21a77c
 0x0000001b (INIT_ARRAYSZ)               32 (bytes)
 0x0000000f (RPATH)                      Library rpath: [$ORIGIN]
 0x00000010 (SYMBOLIC)                   0x0
 0x0000001e (FLAGS)                      SYMBOLIC BIND_NOW
 0x6ffffffb (FLAGS_1)                    Flags: NOW
 0x6ffffff0 (VERSYM)                     0x41ffc
 0x6ffffffc (VERDEF)                     0x44478
 0x6ffffffd (VERDEFNUM)                  1
 0x6ffffffe (VERNEED)                    0x44494
 0x6fffffff (VERNEEDNUM)                 4
 0x00000000 (NULL)                       0x0
```